pkg/kubecost/allocation.go

```diff
@@ -2261,7 +2261,70 @@ func (asr *AllocationSetRange) Accumulate() (*AllocationSet, error) {
 }
 
 // TODO accumulate into lower-resolution chunks of the given resolution
-// func (asr *AllocationSetRange) AccumulateBy(resolution time.Duration) *AllocationSetRange
+func (asr *AllocationSetRange) AccumulateBy(resolution time.Duration) (*AllocationSetRange, error) {
+	allocSetRange := &AllocationSetRange{allocations: []*AllocationSet{}}
+	var allocSet *AllocationSet
+	var err error
+
+	asr.Lock()
+	defer asr.Unlock()
+
+	// use window() with no lock
+	if asr.window().IsEmpty() {
+		return asr, nil
+	}
+
+	// Call total accumulate func if resolution is greater than total window duration
+	if resolution > asr.window().Duration() {
+		// unable to acquire lock here if old asr.accumulate is called
+		for _, as := range asr.allocations {
+			allocSet, err = allocSet.accumulate(as)
+			if err != nil {
+				return nil, err
+			}
+		}
+		allocSetRange.allocations = append(allocSetRange.allocations, allocSet)
+		return allocSetRange, nil
+	}
+
+	asrWindow := asr.window()
+	asrResolution := asrWindow.Duration()
+
+	// if asrResolution < 1 hour then there is not enough data to accumulate
+	// if as.allocations(end - start) > resolution then no need, it's already accumulated. Requery to make more granular in the future?
+	// check this again?
+	if asrResolution <= time.Hour || asrResolution < resolution {
+		return asr, nil
+	}
```

Comment:
I think the `asrResolution <= time.Hour` check shouldn't be here. This is ETL implementation-/configuration-specific information that shouldn't matter for a generic `AccumulateBy()` on a data type.

`asrResolution < resolution` looks like a possibly-unnecessary performance improvement to avoid copying. Does the standard logic not handle this case?

Reply:
I wasn't sure on what our stance was for performance gains vs extra logic. Given that this isn't a function reliant on performance so much I agree. Removed